client/blocks/comment-detail/comment-detail-author.jsx

```diff
@@ -59,9 +70,56 @@ export class CommentDetailAuthor extends Component {
 		gmtOffset( this.props.site )
 	).format( 'll LT' );
 
+	showMoreInfo = () => !! this.props.authorEmail || !! this.props.authorIp || !! this.props.authorUrl;
+
+	toggleBlockUser = () => {
+		const {
+			authorEmail,
+			authorId,
+			authorIsBlocked,
+			siteBlacklist,
+			commentId,
+			translate,
+			updateBlacklist,
+		} = this.props;
+
+		const noticeOptions = {
+			duration: 5000,
+			id: `comment-notice-${ commentId }`,
+			isPersistent: true,
+		};
+
+		const analytics = {
+			action: authorIsBlocked ? 'unblock_user' : 'block_user',
+			user_type: authorId ? 'wpcom' : 'email_only',
+		};
```

Comment:
I think it might be a bit clearer to keep the analytics logic in one place. Perhaps consider passing through `authorId` and `authorIsBlocked` to the updateBlacklist action.

Reply:
But then we'd need to move the logic into `mapDispatchToProps`, making it a bit harder to follow imho.